btsieve/src/ethereum/pattern_matching.rs

```diff
@@ -25,34 +30,45 @@ impl TransactionQuery {
                 transaction_data_length,
                 events,
             } => {
-                let mut result = true;
-
                 if let Some(from_address) = from_address {
-                    result = result && (transaction.from == *from_address);
+                    if transaction.from != *from_address {
+                        return false;
+                    }
                 }
 
                 if let Some(to_address) = to_address {
-                    result = result && (transaction.to == Some(*to_address));
+                    if transaction.to != Some(*to_address) {
+                        return false;
+                    }
                 }
 
                 if let Some(is_contract_creation) = is_contract_creation {
-                    // to_address is None for contract creations
-                    result = result && (*is_contract_creation == transaction.to.is_none());
+                    // transaction.to_address is None iff the transaction creates a contract
```

Comment:
```suggestion
                    // transaction.to_address is None if the transaction creates a contract
```

Reply:
not a typo :) https://www.merriam-webster.com/dictionary/iff

Reply:
Discussed offline - agreed to use full English.